### PR TITLE
Migrate suggestion-composer-tests to typescript

### PR DESCRIPTION
**Pull request by lindner**
- Also cleans up usage of MockSlotComposer
- Adds useful error messages to make it easier to migrate to MockSlotComposer

### CODE REVIEW

File: src/runtime/test/suggestion-composer-tests.ts [hunk 1]
```diff
@@ -9,96 +9,105 @@
...
-  }
-
-  setSuggestions(suggestions) {
-    this.suggestions = suggestions;
+  get suggestConsumers() {
+    return this._suggestConsumers;
   }
 }
 
-describe('suggestion composer', function() {
```
Comment (lindner) [inline src/runtime/test/suggestion-composer-tests.ts]: this test doesn't appear to do much, right?

File: src/runtime/testing/fake-slot-composer.ts [hunk 2]
```diff
@@ -19,7 +19,6 @@ export class FakeSlotComposer extends SlotComposer {
   constructor(options: SlotComposerOptions = {}) {
     super({
       rootContainer: {'root': 'root-context'},
-      modalityName: options.modalityName,
```
Comment (lindner) [inline src/runtime/testing/fake-slot-composer.ts]: already in options, so can be removed...

File: src/runtime/test/suggestion-composer-tests.ts [hunk 3]
```diff
@@ -9,96 +9,105 @@
...
-  constructor() {
-    super(null, new FakeSlotComposer({containers: {suggestions: {}}}));
-    this.suggestions = [];
-    this.updatesCount = 0;
-    this.updateResolve = null;
-  }
-
-  setSuggestions(suggestions) {
-    this.suggestions = suggestions;
+  get suggestConsumers() {
```
Comment (mmandlis) [inline src/runtime/test/suggestion-composer-tests.ts]: maybe better to just have a public getter for `suggestConsumers` in suggestion-composer, to avoid the need to extend the class in test?

File: src/runtime/test/suggestion-composer-tests.ts [hunk 4]
```diff
@@ -9,96 +9,105 @@
...
-
-    await suggestionComposer.setSuggestions([4, 5]);
-    assert.lengthOf(suggestionComposer.suggestions, 2);
-
-    await suggestionComposer.setSuggestions([6, 7, 8]);
-    await suggestionComposer.setSuggestions([]);
-    assert.isEmpty(suggestionComposer.suggestions);
-  });
-
+describe('suggestion composer', () => {
```
Comment (mmandlis) [inline src/runtime/test/suggestion-composer-tests.ts]: just curios, did you delete this test case on purpose?

File: src/runtime/test/suggestion-composer-tests.ts (hunk 3 above)
Comment (lindner) [reply src/runtime/test/suggestion-composer-tests.ts]: I guess the larger question should be, is that internal state necessary to be exposed?  I'm guessing the answer is no; since code works without it.

I'd rephrase the question -- is there a way to replace access to suggestConsumers by looking at the external state of the object only and thus achieve black box testing here...

File: src/runtime/test/suggestion-composer-tests.ts (hunk 4 above)
Comment (lindner) [reply src/runtime/test/suggestion-composer-tests.ts]: See above.

Yes, it doesn't really do anything other than testing the test class setSuggestions method, which doesn't do anything special.

**Comment by lindner** [comment]
Will merge-as-is.  There are more usages of MockSlotComposer to clean up so can address them in a follow on.
